### ISSUE TITLE
MLPClassifier mis-interprets label (multiclass vs multioutput)

### ISSUE THREAD

**Issue by algarnims** [jpmml/sklearn2pmml#376]
Hello,

First, I apprciate your efforts in developing and maintaining the sklearn2pmml repository. This package has been an incredibly useful for us.

I think there is a problem with MLPClassifier. This problem is encountered while trying to save my MLPClassifier model to PMML format using the package. The error message reads as follows:

java.lang.ClassCastException: java.lang.Integer cannot be cast to java.util.List

I have tried to troubleshoot this issue on my own, but I have not been able to find a solution that works. 

Thank you once again for your work on this project.

My environment include: 
Python 3.10.9
scikit-learn 1.2.1
sklearn2pmml 0.92.0

Here is an simple example code:
```python
from sklearn.neural_network import MLPClassifier
from sklearn2pmml import sklearn2pmml, PMMLPipeline
from sklearn.model_selection import train_test_split
import numpy as np

# Data: simple 3 coulmns data with 5 classes 
X = np.random.rand(1000, 3)
y = np.mean(X, axis=1)/2
y=(y*10).astype(np.int64)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# MLPClassifier
model = MLPClassifier(hidden_layer_sizes=(10,5,2), max_iter=1000)
model.fit(X_train, y_train)

# pmml
pmml_pipeline = PMMLPipeline([	
 	("classifier", model)
])
sklearn2pmml(pmml_pipeline, 'model01.pmml')
```

Best regards,

**Comment by vruusmann** [comment]
Appreciate your reproducible example - helps me to see this mysterious `ClassCastException` myself:
```
WARNING: Attribute 'sklearn2pmml.pipeline.PMMLPipeline.target_fields' is not set. Assuming [y1, y2, y3, y4, y5] as the name of target fields
Exception in thread "main" java.lang.ClassCastException: java.lang.Integer cannot be cast to java.util.List
        at sklearn2pmml.pipeline.PMMLPipeline.initLabel(PMMLPipeline.java:581)
        at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:130)
        at com.sklearn2pmml.Main.run(Main.java:91)
        at com.sklearn2pmml.Main.main(Main.java:66)
```

**Comment by vruusmann** [comment]
For an unknown reason, the SkLearn2PMML has trouble figuring out the intent of the classification task. It currently thinks that it's dealing with a multioutput task (ie. five independent labels). The correct thinking would be a multiclass task (a single label with five independent classes).

The problem can be fixed by wrapping the `y(_train)` from Numpy array to Pandas series, and calling `PMMLPipeline.fit(X, y)` with it (instead of `MLPClassifier.fit(X, y)`):

```python
from pandas import Series

# THIS!
y_train = Series(y_train, name = "y")

model = MLPClassifier(hidden_layer_sizes=(10,5,2), max_iter=1000)
# Don't do MLPClassifier.fit(X, y)
#model.fit(X_train, y_train)

pmml_pipeline = PMMLPipeline([	
 	("classifier", model)
])
# THIS!
pmml_pipeline.fit(X_train, y_train)

sklearn2pmml(pmml_pipeline, 'model01.pmml')
``` 

These two changes together help SkLearn2PMML to understand that it's dealing with a multiclass classification problem (one label, five category levels).

**Comment by HighPerformance01** [comment]
Thank you very much for your quick response. We hope this will be resolved in the upcoming versions also which will improve the package overall.

**Comment by algarnims** [comment]
Thank you very much for your quick response. We hope to see this resolved in the future which will improve the package overall.

**Comment by vruusmann** [comment]
The SkLearn2PMML package is fine. It is your pipeline that needs upgrading.

Most importantly, you should be calling `PMMLPipeline.fit(X, y)` always (as opposed to calling `Model.fit(X, y)` and then wrapping it into a dummy `PMMLPipeline` object), because that will set the `PMMLPipeline.active_fields` and `PMMLPipeline.target_fields` attributes to meaningful values. This way, the SkLearn2PMML package does not need to make guesses about the nature of the classification problem.

Look at your own SkLearn2PMML console log that precedes the error. It clearly states that SkLearn2PMML makes an assumption that there must be five labels present (ie. multioutput).

**Comment by vruusmann** [comment]
The fix will probably be in the form of making the "has the PMMLPipeline object been fitted or not" check more strict.

So, instead of making a guess, it would raise a clear "the PMMLPipeline object is not fitted" error.

**Comment by vruusmann** [comment]
The `y_train` dataset is one-dimensional (scalar label), but for some reason the value of the `model.n_outputs_` attribute is 5. This makes SkLearn2PMML/JPMML-SkLearn think that it's dealing with a five-label multioutput problem.

Also, starting from SkLearn2PMML version 0.97.X it is possible to convert fitted Scikit-Learn estimators as-is; there is no need to wrap them to a `PMMLPipeline` anymore:

```python
from sklearn.neural_network import MLPClassifier
from sklearn2pmml import sklearn2pmml

model = MLPClassifier(hidden_layer_sizes=(10,5,2), max_iter=1000)
model.fit(X_train, y_train)

sklearn2pmml(model, "MLPClassifier.pmml")
```
